filter_plugins/oo_filters.py

```diff
@@ -273,16 +276,52 @@ def oo_filter_list(data, filter_attr=None):
                 returns [ { a: 1, b: True },
                           { a: 5, b: True } ]
         '''
-        if not issubclass(type(data), list):
+        if not isinstance(data, list):
             raise errors.AnsibleFilterError("|failed expects to filter on a list")
 
-        if not issubclass(type(filter_attr), str):
-            raise errors.AnsibleFilterError("|failed expects filter_attr is a str")
+        if not isinstance(filter_attr, basestring):
+            raise errors.AnsibleFilterError("|failed expects filter_attr is a str or unicode")
 
         # Gather up the values for the list of keys passed in
         return [x for x in data if x.has_key(filter_attr) and x[filter_attr]]
 
     @staticmethod
+    def oo_nodes_with_label(nodes, label=None, value=None):
+        ''' Filters a list of nodes by label '''
+        if not isinstance(nodes, list):
+            raise errors.AnsibleFilterError("failed expects to filter on a list")
+        if label is None:
+            return nodes
```

Comment:
I was thinking of the case where a variable was used to pass into the filter, but I agree it seems like an issue around misuse.  I'll updated labels to make it a required parameter and remove the return if label is None.